addons/account/models/reconciliation_widget.py

```diff
@@ -488,51 +513,69 @@ def process_move_lines(self, data):
     ####################################################
 
     def _str_domain_for_mv_line(self, search_str):
-        return [
-            '|', ('account_id.code', 'ilike', search_str),
-            '|', ('move_id.name', 'ilike', search_str),
-            '|', ('move_id.ref', 'ilike', search_str),
-            '|', ('date_maturity', 'like', parse_date(self.env, search_str)),
-            '&', ('name', '!=', '/'), ('name', 'ilike', search_str)
-        ]
+        warnings.warn(_('This method is deprecated: %s') % '_str_domain_for_mv_line')
+
+    def _str_query_for_mv_line(self, from_clause, where_clause, where_clause_params, search_str):
+        where_clause += """
+                account.code ILIKE %(search_str)s
+                OR move.name ILIKE %(search_str)s
+                OR move.ref ILIKE %(search_str)s
+                OR (line.name != '/' AND line.name ILIKE %(search_str)s)
+                OR partner.name ILIKE %(search_str)s"""
+        params = {
+            'search_str': '%%%s%%' % search_str,
+        }
+        try:
+            params['date_maturity'] = fields.Date.to_date(parse_date(self.env, search_str))
```

Comment:
Not sure why this is handled differently than before. Usually, silently failing is not a good thing, and it's better to pop up an error.

Reply:
It failled silently before too. I have to catch the error now because I have to convert it to `datetime.date` to pass it as a parameter to `psycopg2`'s `query` function. If the it wasn't well formatted, it was ignored before too.